crates/hyperdrive-math/src/long/open.rs

```diff
@@ -107,50 +107,50 @@ impl State {
 
     /// Calculate an updated pool state after opening a long.
     ///
-    /// For a given base amount and bond amount, base is converted to
+    /// For a given base delta and bond delta, the base delta is converted to
     /// shares and the reserves are updated such that
-    /// `state.bond_reserves -= bond_amount` and
-    /// `state.share_reserves += base_amount / vault_share_price`.
+    /// `state.bond_reserves -= bond_delta` and
+    /// `state.share_reserves += base_delta / vault_share_price`.
     pub fn calculate_pool_state_after_open_long(
         &self,
         base_amount: FixedPoint,
-        maybe_bond_amount: Option<FixedPoint>,
+        maybe_bond_deltas: Option<FixedPoint>,
     ) -> Result<Self> {
-        let bond_amount = match maybe_bond_amount {
-            Some(bond_amount) => bond_amount,
-            None => self.calculate_open_long(base_amount)?,
-        };
+        let (share_deltas, bond_deltas) =
+            self.calculate_pool_share_bond_deltas_after_open_long(base_amount, maybe_bond_deltas)?;
         let mut state = self.clone();
-        state.info.bond_reserves -= bond_amount.into();
-        state.info.share_reserves += (base_amount / self.vault_share_price()).into();
+        state.info.bond_reserves -= bond_deltas.into();
+        state.info.share_reserves += share_deltas.into();
         Ok(state)
     }
 
     /// Calculate the share deltas to be applied to the pool after opening a long.
-    pub fn calculate_pool_deltas_after_open_long(
+    pub fn calculate_pool_share_bond_deltas_after_open_long(
```

Comment:
I actually think the old name with the new return value match well. `pool_share_bond_deltas` is pretty hard to interpret.
```suggestion
    pub fn calculate_pool_deltas_after_open_long(
```